### PR TITLE
Add support for types 8, 9 & 10

### LINKED ISSUE #17: DumpType = 9
In https://github.com/0vercl0k/wtf/issues/101, there is a dump file with a `DumpType = 9` which isn't supported by `kdmp-parser`. Based on the investigation, this seems to be newish and only available in latest (?) WinDbgX.

### LINKED ISSUE #18: DumpType = 0xA, Kernel range dump
Based on https://github.com/0vercl0k/wtf/issues/139 this looks like yet another new dump format:
```
kd> dx @$cursession
@$cursession                 : 64-bit Kernel range dump: testapps\state\mem.dmp
```

It doesn't seem to be supported by the `dbgeng.dll` shipped w/ the regular windbg that I got in the SDK:
```
0:000> lmvm dbgeng
Browse full module list
start             end                 module name
00000001`80000000 00000001`8087a000   dbgeng     (pdb symbols)          c:\work\dbg\sym\dbgeng.pdb\DA8D57515A772495F39B6FECD19C2C8D1\dbgeng.pdb
    Loaded symbol image file: dbgeng.dll
    Mapped memory image file: c:\program Files (x86)\windows kits\10\debuggers\x64\dbgeng.dll
    Image path: c:\program Files (x86)\windows kits\10\debuggers\x64\dbgeng.dll
    Image name: dbgeng.dll
    Browse all global symbols  functions  data
    Image was built with /Brepro flag.
    Timestamp:        2249EE61 (This is a reproducible build file hash, not a timestamp)
    CheckSum:         0084EADC
    ImageSize:        0087A000
    File version:     10.0.22621.1
    Product version:  10.0.22621.1
    File flags:       0 (Mask 3F)
    File OS:          40004 NT Win32
    File type:        2.0 Dll
    File date:        00000000.00000000
    Translations:     0409.04b0
    Information from resource tables:
        CompanyName:      Microsoft Corporation
        ProductName:      Microsoft® Windows® Operating System
        InternalName:     DbgEng.Dll
        OriginalFilename: DbgEng.Dll
        ProductVersion:   10.0.22621.1
        FileVersion:      10.0.22621.1 (WinBuild.160101.0800)
        FileDescription:  Windows Symbolic Debugger Engine
        LegalCopyright:   © Microsoft Corporation. All rights reserved.
```

### PR DESCRIPTION
**Pull request by hugsy**
## Improvements

Add support for dump types for modern WinDbg versions:
 - [x] 0x8 -> `.dump /k` (Create a dump with kernel memory only)
 - [x] 0x9 -> `.dump /ka` (Create a dump with active kernel and user mode memory)
 - [x] 0xa -> `.dump /f` (Create a full dump)

Also:
 - Version bumped 0.6 -> 0.7
 - Adds `FizeSize()` to the FileMap
 - Exposes the `HEADER64` to Python too, as justly noted by @neitsa this simply addition is costless, but very valuable.

## Additional notes

 1. Although the C++ API was not changed, some implementation details had to be changed. Please let me know if you disagree;
 2. The new format offsets were found through reversing, hence some fields `__unknownXX`. However I tested against quite a number of dumps, and could validate my findings but since there's no public implementation (kdmp-parser will be the first 😉) I couldn't fact-check it otherwise;
 3. The new testcases [have been commented](https://github.com/hugsy/kdmp-parser/pull/1/files#diff-f65f2a7212a3b848c7883fceae0601a47c705fc072759758a618889315815600R176-R177) because they will fail until we provide a correct URL to download the new testdatas ZIP 
 4. You'll see blocks like this

```c
#pragma GCC diagnostic push
#pragma GCC diagnostic ignored "-Winvalid-offsetof"
[...]
#pragma GCC diagnostic pop
```

Those are GCC specific errors claiming not to be able to calculate the offset checks used in the `static_assert`. This is completely wrong, and the other compilers (CLANG/CL) don't say anything. In addition, even GCC produces the correct format so the behavior doesn't change in any way. Therefore I chose to explicitly silence it (punctually to those `assert` checks).

Happy to discuss any of those points.

Fixes #17 
Fixes #18

### CODE REVIEW

**Comment by 0vercl0k** [comment]
Booyah!

Will look at this this week, thanks again for your hard work!

Cheers

File: src/tests/tests_parser.cc [hunk 1]
```diff
@@ -173,8 +173,8 @@ const static TestCaseValues TestCaseCompleteDump{
 
 const std::array g_ExpectedValues{
     TestCaseBmp, TestCaseFull,
-    // TestCaseKernelDump,   TestCaseKernelUserDump,
-    // TestCaseCompleteDump,
+    TestCaseKernelDump,   TestCaseKernelUserDump,
```
Comment (hugsy) [inline src/tests/tests_parser.cc:176]: 🎉

Comment (0vercl0k) [reply src/tests/tests_parser.cc:176]: Hehe yes! I can start code-reviewing now!

**Comment by 0vercl0k** [comment]
Ok I did first pass, I think it's looking good. I need to do a second pass and check the misc stuff, like small documentation for the Python README and then we should be good to go!

Cheers

File: CMakeLists.txt [hunk 2]
```diff
@@ -11,14 +11,14 @@ project(
...
+    VERSION 0.7.0
 )
 
 set(PROJECT_AUTHOR 0vercl0k)
 set(PROJECT_LICENSE MIT)
 
 option(BUILD_PARSER "Build the parser executable for kdmp-parser" ON)
-option(BUILD_TESTS "Build the test suite for kdmp-parser" ON)
+option(BUILD_TESTS "Build the test suite for kdmp-parser" OFF)
 option(BUILD_PYTHON_BINDING "Build the Python bindings for kdmp-parser" ON)
```
Comment (hugsy) [inline CMakeLists.txt:22]: Quick suggestion, set this option to `OFF` by default? Building the bindings requires more dependencies they may not want/need - and I assume most people who only want the bindings will go through PyPI.

```suggestion
option(BUILD_PYTHON_BINDING "Build the Python bindings for kdmp-parser" OFF)
```

Comment (0vercl0k) [reply CMakeLists.txt]: Good catch!

**Comment by hugsy** [comment]
Thanks for the review & merge !

**Pull request merged**